lib/saluki-io/src/net/util/retry/queue/persisted.rs

```diff
@@ -92,6 +134,14 @@ where
         self.entries.len()
     }
 
+    fn on_disk_bytes_limit(&self) -> Result<u64, GenericError> {
```

Comment:
It would be nice to just give this a doc comment explaining that we use this function to determine if writing to disk would cause the underlying volume to end up with more than `self.storage_max_disk_ratio` in terms of used space, even if we haven't yet hit `self.max_on_disk_bytes`.